### PR TITLE
feat(grouping): Add support for automatic grouping updates

### PR DESCRIPTION
**Pull request by mitsuhiko**
This implements automatic grouping updates. The feature is currently still disabled so we can work on messaging and allow users to opt out.

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -585,9 +587,32 @@ def save(
...
+def _auto_update_grouping(project):
+    if not settings.SENTRY_GROUPING_AUTO_UPDATE_ENABLED:
+        return
+
+    old_grouping = project.get_option("sentry:grouping_config")
+    new_grouping = DEFAULT_GROUPING_CONFIG
+
+    # update to latest grouping config but not if a user is already on
+    # beta.
+    if old_grouping != new_grouping and old_grouping != BETA_GROUPING_CONFIG:
```
Comment (markstory) [inline src/sentry/event_manager.py:607]: How do orgs leave beta configs automatically?

**Review by markstory — COMMENTED** [review]
It would be good to have some tests for the API endpoint behavior and the grouping algorithm upgrades on save_event

File: src/sentry/event_manager.py (hunk 1 above)
Comment (mitsuhiko) [reply src/sentry/event_manager.py:607]: At the moment there is only one beta config and we just want to remove this thing entirely going forward. The future of hierarchical grouping (which is the beta grouping) is a bit unclear at the moment. So we will manually transition them out once we have a path forward and then retire the beta grouping concept.

**Review by untitaker — COMMENTED** [review]
I think it would be "saner" if grouping auto-upgrade didn't immediately happen upon deploy but e.g. on a fixed day of the week.

**Comment by mitsuhiko** [comment]
@untitaker it should not take place until the setting flag is flipped. So we have time to communicate before flipping.

**Comment by untitaker** [comment]
okay, i mean if we create a new grouping algorithm while this setting is already enabled

**Comment by mitsuhiko** [comment]
I think we can ignore that case. We won't make a new one in the interim anyways.

**Review by Swatinem — APPROVED** [review]
lgtm, though do you really want to opt-in all the existing projects to auto updates? Or just enable auto-updates for new projects?

**Pull request merged**